src/components/textarea/useResizeTextAreaOnChange.ts

```diff
@@ -0,0 +1,31 @@
+import syncTextAreaHeight from './syncTextAreaHeight'
+
+import type { ChangeEvent, ChangeEventHandler, RefObject } from 'react'
+
+type UseResizeTextAreaOnChangeConfig = {
+  isEnabled: boolean
+  shadowTextAreaRef: RefObject<HTMLTextAreaElement>
+  textAreaRef: RefObject<HTMLTextAreaElement>
+}
+
+type UseResizeTextAreaOnChangeDecorator = (onChange?: ChangeEventHandler) => ChangeEventHandler
+
+export default function useResizeTextAreaOnChange({
+  isEnabled,
+  shadowTextAreaRef,
+  textAreaRef,
+}: UseResizeTextAreaOnChangeConfig): UseResizeTextAreaOnChangeDecorator {
+  return function decorateTextAreaChangeHandler(onChange?: ChangeEventHandler) {
+    return function syncTextAreaHeightAfterOnChange(event: ChangeEvent) {
+      onChange?.(event)
```

Comment:
**note:** As the code comment below points out, we allow the consumer-supplied `onChange` handler to call `event.preventDefault()` to stop our resizing behaviour from happening.